62_CAD/DrawResourcesFiller.h

```diff
@@ -120,6 +122,36 @@ struct DrawResourcesFiller
 				geometryInfo.getAlignedStorageSize();
 		}
 	};
+
+	struct IGeoreferencedImageLoader : IReferenceCounted
+	{
+		virtual core::smart_refctd_ptr<ICPUBuffer> load(std::filesystem::path imagePath, uint32_t2 offset, uint32_t2 extent, uint32_t mipLevel) = 0;
+
+		virtual uint32_t2 getExtents(std::filesystem::path imagePath) = 0;
+
+		virtual asset::E_FORMAT getFormat(std::filesystem::path imagePath) = 0;
+	};
+
+	void setGeoreferencedImageLoader(core::smart_refctd_ptr<IGeoreferencedImageLoader>&& _georeferencedImageLoader)
+	{
+		georeferencedImageLoader = _georeferencedImageLoader;
+	}
+
+	uint32_t2 queryGeoreferencedImageExtents(std::filesystem::path imagePath)
+	{
+		return georeferencedImageLoader->getExtents(imagePath);
+	}
+
+	asset::E_FORMAT queryGeoreferencedImageFormat(std::filesystem::path imagePath)
+	{
+		return georeferencedImageLoader->getFormat(imagePath);
+	}
+
+	// These are vulkan standard, might be different in n4ce!
+	constexpr static float64_t3 topLeftViewportNDC = float64_t3(-1.0, -1.0, 1.0);
+	constexpr static float64_t3 topRightViewportNDC = float64_t3(1.0, -1.0, 1.0);
+	constexpr static float64_t3 bottomLeftViewportNDC = float64_t3(-1.0, 1.0, 1.0);
+	constexpr static float64_t3 bottomRightViewportNDC = float64_t3(1.0, 1.0, 1.0);
```

Comment:
I like that you documented your assumptions, we always use vulkan standard.

we will probably need to document these assumptions with a more "Global" settings.

After giving it some thought, I think the ndc standard choice should be documented for Nabla. (yes we use Vulkan internally) but  I think it's Nabla's job to decide and force a standard.

So my suggestion is move these locally to the function you're using them in and write a single comment about this assumption. It's no biggie